openpype/plugins/publish/integrate.py

```diff
@@ -530,15 +542,33 @@ def prepare_representation(self, repre,
         template_data["representation"] = repre["name"]
         template_data["ext"] = repre["ext"]
 
+        stagingdir = repre.get("stagingDir")
+        if not stagingdir:
+            # Fall back to instance staging dir if not explicitly
+            # set for representation in the instance
+            self.log.debug((
+                "Representation uses instance staging dir: {}"
+            ).format(instance_stagingdir))
+            stagingdir = instance_stagingdir
+
+        if not stagingdir:
+            raise KnownPublishError(
+                "No staging directory set for representation: {}".format(repre)
+            )
+
         # optionals
         # retrieve additional anatomy data from representation if exists
+        if not instance.data.get("originalDirname"):
```

Comment:
I would say original dirname and basename is representation specific and not instance specific.

Reply:
Ok, used `instance_stagingdir` instead.